agent/operator_endpoint.go

```diff
@@ -265,23 +266,50 @@ func (s *HTTPServer) OperatorAutopilotConfiguration(resp http.ResponseWriter, re
 	}
 }
 
-// FixupConfigDurations is used to handle parsing the duration fields in
-// the Autopilot config struct
-func FixupConfigDurations(raw interface{}) error {
+type durationFixer map[string]struct{}
+
+func NewDurationFixer(fields ...string) durationFixer {
+	d := make(map[string]struct{})
```

Comment:
@magiconair would say to make this a `map[string]bool` and skip the `ok` checks to simplify the code.